src/compiler/parser.ts

```diff
@@ -2171,7 +2198,7 @@ namespace ts {
         }
 
         function isStartOfParameter(): boolean {
-            return token() === SyntaxKind.DotDotDotToken || isIdentifierOrPattern() || isModifierKind(token()) || token() === SyntaxKind.AtToken || token() === SyntaxKind.ThisKeyword;
+            return token() === SyntaxKind.DotDotDotToken || isIdentifierOrPattern() || isModifierKind(token()) || token() === SyntaxKind.AtToken || token() === SyntaxKind.ThisKeyword || token() === SyntaxKind.NewKeyword;

```

Comment:
At least one test entered an infinite loop with `isStartOfType` inside `isStartOfParameter`. Instead I just added `StringLiteral` and `NumericLiteral` to the end.